src/util.c

```diff
@@ -45,3 +46,40 @@ void *ecalloc(size_t nmemb, size_t size)
         err(EXIT_FAILURE, "calloc");
     return p;
 }
+
+void *erealloc(void *ptr, size_t size)
+{
+    void *p = realloc(ptr, size);
+    if (!p)
+        err(EXIT_FAILURE, "realloc");
+    return p;
+}
+
+void streamReserve(Stream *buf, size_t size)
+{
+    assert(buf->off <= buf->cap);
+    size_t avail = buf->cap - buf->off;
+    if (avail < size) {
+        buf->cap += MAX(size, 128); /* be a bit greedy when allocating */
+        buf->buf = erealloc(buf->buf, buf->cap);
+    }
+    assert((buf->off + size) <= buf->cap);
+}
+
+void streamChar(Stream *buf, char ch)
+{
+    streamReserve(buf, 1);
+    buf->buf[buf->off++] = ch;
+}
+
+void streamMem(Stream *buf, const char *str, size_t n)
+{
+    streamReserve(buf, n);
+    for (size_t i = 0; i < n; ++i)
+        buf->buf[buf->off++] = str[i];
+}
```

Comment:
I just noticed, we have `memcpy()` to copy memory. It's even likely this code gets optimized into a `memcpy()` call. And the 2nd parameter's type should be `void *` to show we're dealing with any memory.

Reply:
Did it this way to keep the copy and `off` in sync in the same line. But I don't have strong preferences here, can change it to `memcpy`.

Reply:
Fixed in new push.